torch/csrc/jit/frontend/schema_matching.cpp

```diff
@@ -542,12 +624,39 @@ std::pair<size_t, MatchedSchema> matchSchemas(
           kwargs,
           self,
           render_errors ? &failure_messages : nullptr,
-          allow_conversions);
+          allow_conversions,
+          /*allow_narrowing_conversions=*/false);
       if (matched_schema) {
         return std::make_pair(i, *matched_schema);
       }
     }
   }
+
+  // Now we try narrowing conversions
+  std::vector<std::pair<size_t, MatchedSchema>> matches;
+  for (size_t i = 0; i < schemas.size(); ++i) {
+    const auto matched_schema = tryMatchSchema(
+        *schemas[i],
+        loc,
+        graph,
+        args,
+        kwargs,
+        self,
+        render_errors ? &failure_messages : nullptr,
+        /*allow_conversions=*/true,
+        /*allow_narrowing_conversions=*/true);
```

Comment:
https://github.com/pytorch/pytorch/pull/57466/files#diff-be55eef0e7c14230bf8bcd3bbbf23193e981bf3138c57092ccfee1c565a61302L532-R647 - Since only `allow_conversions` and `allow_narrowing_conversions` are being changed, is it possible to combine both the calls? Like using a similar loop for `allow_narrowing_conversions` ?

Reply:
Hmm, maybe. We only want to allow narrowing conversions if we find a single matching schema, which is why we have to try all the matches with `allow_narrowing_conversions` and see how many we get. On the other hand, the original logic (starting at 609 with `for (bool allow_conversions : {false, true})...`) greedily picks the first matching schema. Also, `allow_narrowing_conversions=true` shouldn't happen if `allow_conversions=false`, so we should skip that permutation. I could combine the two, but I think having so many conditional checks should get messy.